src/thermo/PengRobinsonMFTP.cpp

```diff
@@ -0,0 +1,1310 @@
+    alpha_vec_Curr_[k] = sqt_alpha*sqt_alpha;
+}
+
+void PengRobinsonMFTP::setSpeciesCoeffs(const std::string& species,
+                                        double a, double b, double w)
+{
+    size_t k = speciesIndex(species);
+    if (k == npos) {
+        throw CanteraError("PengRobinsonMFTP::setSpeciesCoeffs",
+            "Unknown species '{}'.", species);
+    }
+    size_t counter = k + m_kk * k;
+    a_coeff_vec(0, counter) = a;
+    // we store this locally because it is used below to calculate a_Alpha:
+    double aAlpha_k = a*alpha_vec_Curr_[k];
+    aAlpha_coeff_vec(0, counter) = aAlpha_k;
+
+    // standard mixing rule for cross-species interaction term
+    for (size_t j = 0; j < m_kk; j++) {
+        if (k == j) {
+            continue;
+        }
+        double a0kj = sqrt(a_coeff_vec(0, j + m_kk * j) * a);
+        double aAlpha_j = a*alpha_vec_Curr_[j];
+        double a_Alpha = sqrt(aAlpha_j*aAlpha_k);
+        if (a_coeff_vec(0, j + m_kk * k) == 0) {
+            a_coeff_vec(0, j + m_kk * k) = a0kj;
+            aAlpha_coeff_vec(0, j + m_kk * k) = a_Alpha;
+            a_coeff_vec(0, k + m_kk * j) = a0kj;
+            aAlpha_coeff_vec(0, k + m_kk * j) = a_Alpha;
+        }
+    }
+    a_coeff_vec.getRow(0, a_vec_Curr_.data());
+    aAlpha_coeff_vec.getRow(0, aAlpha_vec_Curr_.data());
+    b_vec_Curr_[k] = b;
+}
+
+void PengRobinsonMFTP::setBinaryCoeffs(const std::string& species_i,
+        const std::string& species_j, double a0, double alpha)
+{
+    size_t ki = speciesIndex(species_i);
+    if (ki == npos) {
+        throw CanteraError("PengRobinsonMFTP::setBinaryCoeffs",
+            "Unknown species '{}'.", species_i);
+    }
+    size_t kj = speciesIndex(species_j);
+    if (kj == npos) {
+        throw CanteraError("PengRobinsonMFTP::setBinaryCoeffs",
+            "Unknown species '{}'.", species_j);
+    }
+
+    size_t counter1 = ki + m_kk * kj;
+    size_t counter2 = kj + m_kk * ki;
+    a_coeff_vec(0, counter1) = a_coeff_vec(0, counter2) = a0;
+    aAlpha_coeff_vec(0, counter1) = aAlpha_coeff_vec(0, counter2) = a0*alpha;
+    a_vec_Curr_[counter1] = a_vec_Curr_[counter2] = a0;
+    aAlpha_vec_Curr_[counter1] = aAlpha_vec_Curr_[counter2] = a0*alpha;
+}
+
+// ------------Molar Thermodynamic Properties -------------------------
+
+double PengRobinsonMFTP::enthalpy_mole() const
+{
+    _updateReferenceStateThermo();
+    double h_ideal = RT() * mean_X(m_h0_RT);
+    double h_nonideal = hresid();
+    return h_ideal + h_nonideal;
+}
+
+double PengRobinsonMFTP::entropy_mole() const
+{
+    _updateReferenceStateThermo();
+    double sr_ideal = GasConstant * (mean_X(m_s0_R) - sum_xlogx()
+                                                - std::log(pressure()/refPressure()));
+    double sr_nonideal = sresid();
+    return sr_ideal + sr_nonideal;
+}
+
+double PengRobinsonMFTP::cp_mole() const
+{
+    _updateReferenceStateThermo();
+    double TKelvin = temperature();
```

Comment:
I'm going to disagree with @bryanwweber on this - I think "T" is a more clear variable name than "TKelvin".

Reply:
Hmmm... despite the sense of my comment, I actually also have a preference for `T`. My comment was more so to be consistent throughout the file, but I should have been more clear with my preference!

Reply:
Fixed.